### PR TITLE
Refactor waku protocols and utils

### PR DESCRIPTION
**Pull request by staheri14**
This PR is progress towards 
- https://github.com/status-im/nim-waku/issues/639
-  https://github.com/status-im/nim-waku/issues/287.

I have reviewed all the modules under `waku/v2/protocol` and `waku/v2/utils` and made sure they comply with the following items of [nim-waku refactor week: Track 1 checklist](https://hackmd.io/1imOGULZRsed2HpgmzGleA?view).

- [x] Top-level {.push raises: [Defect].} (except for the Swap module, it throws some exceptions which I needed more time to understand their hierarchy and resolve them properly)
- [x] No result returns
- [x] logscope defined for each module that uses chronicles logging
- [x] new() instead of init() for constructors of ref object types

**Next**
Will focus on cleaning imports and better exception handling

### CODE REVIEW

File: waku/v2/protocol/waku_filter/waku_filter.nim [hunk 1]
```diff
@@ -182,11 +189,9 @@ method init*(wf: WakuFilter) =
   wf.codec = WakuFilterCodec
 
 proc init*(T: type WakuFilter, peerManager: PeerManager, rng: ref BrHmacDrbgContext, handler: MessagePushHandler): T =
-  new result
-  result.rng = crypto.newRng()
-  result.peerManager = peerManager
-  result.pushHandler = handler
-  result.init()
+  var wf = WakuFilter(rng: crypto.newRng(), peerManager: peerManager, pushHandler: handler)
```
Comment (oskarth) [inline waku/v2/protocol/waku_filter/waku_filter.nim:192]: Nitpick: stylistic thing but personally prefer to have inits like `crypto.newRng()` in a temp variable. Sometimes makes a difference when it comes to debugging. Also makes line <80ch

File: waku/v2/protocol/waku_lightpush/waku_lightpush.nim [hunk 2]
```diff
@@ -97,17 +103,14 @@ proc init*(T: type PushRPC, buffer: seq[byte]): ProtoResult[T] =
...
 # Protocol -------------------------------------------------------
 proc init*(T: type WakuLightPush, peerManager: PeerManager, rng: ref BrHmacDrbgContext, handler: PushRequestHandler, relay: WakuRelay = nil): T =
   debug "init"
-  new result
-  result.rng = crypto.newRng()
-  result.peerManager = peerManager
-  result.requestHandler = handler
-  result.relayReference = relay
-  result.init()
+  var wl = WakuLightPush(rng: crypto.newRng(), peerManager: peerManager, requestHandler: handler, relayReference: relay)
```
Comment (oskarth) [inline waku/v2/protocol/waku_lightpush/waku_lightpush.nim:111]: Ditto above

File: waku/v2/protocol/waku_message.nim [hunk 3]
```diff
@@ -4,7 +4,7 @@
 ## for spec.
 ##
 ## For payload content and encryption, see waku/v2/node/waku_payload.nim
-
+{.push raises: [Defect].}
```
Comment (oskarth) [inline waku/v2/protocol/waku_message.nim:7]:
```suggestion

{.push raises: [Defect].}
```

File: waku/v2/protocol/waku_rln_relay/rln.nim [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # this module contains the Nim wrappers for the rln library https://github.com/kilic/rln/blob/3bbec368a4adc68cd5f9bfae80b17e1bbb4ef373/src/ffi.rs
-
+{.push raises: [Defect].}
```
Comment (oskarth) [inline waku/v2/protocol/waku_rln_relay/rln.nim:2]:
```suggestion

{.push raises: [Defect].}

```

Comment (oskarth) [reply waku/v2/protocol/waku_rln_relay/rln.nim:2]: Look how this poor statement is crushed between a comment and the import!

File: waku/v2/protocol/waku_swap/waku_swap.nim [hunk 5]
```diff
@@ -20,6 +20,7 @@
 ##
 ## Things like settlement is for future work.
 ##
+# {.push raises: [Defect].}
```
Comment (oskarth) [inline waku/v2/protocol/waku_swap/waku_swap.nim:23]:
```suggestion

# {.push raises: [Defect].}
```

Could you add a brief comment on the immediate error you noticed here?

File: waku/v2/protocol/waku_swap/waku_swap.nim [hunk 6]
```diff
@@ -263,13 +270,9 @@ proc init*(wakuSwap: WakuSwap) =
...
 proc init*(T: type WakuSwap, peerManager: PeerManager, rng: ref BrHmacDrbgContext, swapConfig: SwapConfig): T =
   info "wakuSwap init 2"
-  new result
-  result.rng = rng
-  result.peerManager = peerManager
-  result.accounting = initTable[PeerId, int]()
-  result.text = "test"
-  result.config = swapConfig
-  result.init()
+  var ws = WakuSwap(rng: rng, peerManager: peerManager, accounting: initTable[PeerId, int](), text: "test", config: swapConfig)
```
Comment (oskarth) [inline waku/v2/protocol/waku_swap/waku_swap.nim:273]: Ditto above re table, also line break/format

File: waku/v2/utils/namespacing.nim [hunk 7]
```diff
@@ -31,7 +31,7 @@ proc fromString*(T: type NamespacedTopic, topic: string): NamespacingResult[Name
     # Ensures that topic starts with a "/"
     return err("invalid topic format")
   
-  ok(NamespacedTopic(application: parts[1],
+  return ok(NamespacedTopic(application: parts[1],
```
Comment (oskarth) [inline waku/v2/utils/namespacing.nim:34]: Pull out in Temoporary variable and return `ok` of that

**Review by oskarth — APPROVED** [review]
Thanks! Small comments, mostly nitpicks

File: waku/v2/protocol/waku_filter/waku_filter.nim [hunk 8]
```diff
@@ -111,13 +114,15 @@ proc init*(T: type FilterRequest, buffer: seq[byte]): ProtoResult[T] =
   for buf in buffs:
     rpc.contentFilters.add(? ContentFilter.init(buf))
 
-  ok(rpc)
+  return ok(rpc)
 
 proc encode*(push: MessagePush): ProtoBuffer =
-  result = initProtoBuffer()
+  var output: ProtoBuffer  = initProtoBuffer()
```
Comment (jm-clius) [inline waku/v2/protocol/waku_filter/waku_filter.nim:120]: No strong feelings about this, but I think the briefer
```suggestion
  var output  = initProtoBuffer()

```
is more idiomatic

File: waku/v2/protocol/waku_lightpush/waku_lightpush.nim [hunk 9]
```diff
@@ -34,10 +34,12 @@ const
 
 # Encoding and decoding -------------------------------------------------------
 proc encode*(rpc: PushRequest): ProtoBuffer =
-  result = initProtoBuffer()
+  var output: ProtoBuffer = initProtoBuffer()
```
Comment (jm-clius) [inline waku/v2/protocol/waku_lightpush/waku_lightpush.nim:37]:
```suggestion
  var output = initProtoBuffer()
```

File: waku/v2/protocol/waku_lightpush/waku_lightpush.nim [hunk 10]
```diff
@@ -52,13 +54,15 @@ proc init*(T: type PushRequest, buffer: seq[byte]): ProtoResult[T] =
   discard ? pb.getField(2, buf)
   rpc.message = ? WakuMessage.init(buf)
 
-  ok(rpc)
+  return ok(rpc)
 
 proc encode*(rpc: PushResponse): ProtoBuffer =
-  result = initProtoBuffer()
+  var output: ProtoBuffer = initProtoBuffer()
```
Comment (jm-clius) [inline waku/v2/protocol/waku_lightpush/waku_lightpush.nim:60]:
```suggestion
  var output = initProtoBuffer()
```

File: waku/v2/protocol/waku_lightpush/waku_lightpush.nim [hunk 11]
```diff
@@ -72,14 +76,16 @@ proc init*(T: type PushResponse, buffer: seq[byte]): ProtoResult[T] =
   discard ? pb.getField(2, info)
   rpc.info = info
 
-  ok(rpc)
+  return ok(rpc)
 
 proc encode*(rpc: PushRPC): ProtoBuffer =
-  result = initProtoBuffer()
+  var output: ProtoBuffer = initProtoBuffer()
```
Comment (jm-clius) [inline waku/v2/protocol/waku_lightpush/waku_lightpush.nim:82]:
```suggestion
  var output = initProtoBuffer()
```

**Review by jm-clius — APPROVED** [review]
LGTM! Thanks, @staheri14!

File: waku/v2/utils/namespacing.nim (hunk 7 above)
Comment (staheri14) [reply waku/v2/utils/namespacing.nim:34]: done!

File: waku/v2/protocol/waku_swap/waku_swap.nim (hunk 5 above)
Comment (staheri14) [reply waku/v2/protocol/waku_swap/waku_swap.nim:24]: Sure, done!

File: waku/v2/protocol/waku_swap/waku_swap.nim (hunk 6 above)
Comment (staheri14) [reply waku/v2/protocol/waku_swap/waku_swap.nim:274]: sure!done!

File: waku/v2/protocol/waku_filter/waku_filter.nim (hunk 8 above)
Comment (staheri14) [reply waku/v2/protocol/waku_filter/waku_filter.nim:120]: I know! but personally, I like the explicit type declaration, which makes debugging much easier, and overall enhances the code readability. :) 
I will follow the idiomatic way though!

File: waku/v2/protocol/waku_filter/waku_filter.nim (hunk 1 above)
Comment (staheri14) [reply waku/v2/protocol/waku_filter/waku_filter.nim]: Makes sense! 👍

File: waku/v2/protocol/waku_lightpush/waku_lightpush.nim (hunk 2 above)
Comment (staheri14) [reply waku/v2/protocol/waku_lightpush/waku_lightpush.nim]: done

File: waku/v2/protocol/waku_swap/waku_swap.nim (hunk 5 above)
Comment (staheri14) [reply waku/v2/protocol/waku_swap/waku_swap.nim:29]: I explained the initial error that I noticed, once I tried to resolve it by throwing the exception from the procs, further issues appeared that made me decide to deal with them in a separate PR

Comment (jm-clius) [reply waku/v2/protocol/waku_swap/waku_swap.nim:29]: Yeah, sympathy - you start with a simple `raises` and end up having to go down a rabbit hole in referenced modules.

Comment (staheri14) [reply waku/v2/protocol/waku_swap/waku_swap.nim:29]: Thanks! ;) :D

**Pull request merged**